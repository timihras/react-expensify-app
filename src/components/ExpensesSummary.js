import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({expenseCount, expenseTotal}) => {
  return (
    <div>
      <h1>{`Viewing ${expenseCount} ${expenseCount === 1 ? 'expense' : 'expenses'} totalling ${numeral(expenseTotal / 100).format('$0,0.00')}.`}</h1>
    </div>
  )
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expenseTotal: selectExpensesTotal(visibleExpenses)
  };
;}

export default connect(mapStateToProps)(ExpensesSummary);
