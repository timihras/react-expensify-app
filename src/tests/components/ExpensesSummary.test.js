import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with one expense', () =>{
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={1214} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expenseTotal={161531215} />);
  expect(wrapper).toMatchSnapshot();
});