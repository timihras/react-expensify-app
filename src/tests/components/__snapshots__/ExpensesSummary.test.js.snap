// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 23 expenses totalling $1,615,312.15.
  </h1>
</div>
`;

exports[`should render ExpensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 1 expense totalling $12.14.
  </h1>
</div>
`;
